heat/core/linalg/basics.py

```diff
@@ -44,6 +45,111 @@
 ]
 
 
+def cholesky(x: DNDarray, upper: bool = False) -> DNDarray:
+    """
+    Returns the Cholesky decomposition of a Hermitian or symmetric positive-definite matrix.
```

Comment:
add a note saying this doesnt work for complex values (also change the examples)